tests/integration/test_cos.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import pytest
+
+import charmed_kubeflow_chisme.testing
+from charmed_kubeflow_chisme.testing import (
+    GRAFANA_AGENT_APP,
+    assert_alert_rules,
+    assert_logging,
+    assert_metrics_endpoints,
+    deploy_and_assert_grafana_agent,
+)
+
+# Note(rgildein): Change app metrics endpoint, since blackbox-exporter-k8s is not using
+# 'metrics-endpoint'.
+charmed_kubeflow_chisme.testing.cos_integration.APP_METRICS_ENDPOINT = "self-metrics-endpoint"
+
+
+@pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
+async def test_build_and_deploy(ops_test):
+    """Test deployment of grafana-agent-k8s and relate it with charm."""
+    await ops_test.model.deploy("blackbox-exporter-k8s", channel="latest/stable", trust=True)
```

Comment:
+1 on using blackbox instead of MLflow

Reply:
We can switch to other, when we add logging and metrics to it.